pkg/scheduler/schedule_one.go

```diff
@@ -450,7 +450,8 @@ func (sched *Scheduler) findNodesThatFitPod(ctx context.Context, fwk framework.F
 		return nil, diagnosis, err
 	}
 	// Run "prefilter" plugins.
-	preRes, s := fwk.RunPreFilterPlugins(ctx, state, pod)
+	preRes, s, unScheduledPlugins := fwk.RunPreFilterPlugins(ctx, state, pod)
+	diagnosis.UnschedulablePlugins = unScheduledPlugins
```

Comment:
When all Nodes are filtered out with PreFilterResult, the returning status would be non-success and `diagnosis.AddPluginStatus(s)` at L469 overwrites `UnschedulablePlugins` we set here.

Reply:
So, to clarify, the requirement here is

- set `unscheduledPlugins` to `diagnosis.UnschedulablePlugins` when PreFilter plugin(s) rejected some Nodes or all Nodes with PreFilterResult(s).
- use `diagnosis.AddPluginStatus(s)` to set `diagnosis.UnschedulablePlugins` or `diagnosis.PendingPlugins` when PreFilter plugin rejected the Pod with non-success status.

And, the problem with the current code base is, when PreFilter plugin(s) rejected all Nodes with PreFilterResult(s), we use `diagnosis.AddPluginStatus(s)`; we should use `unscheduledPlugins` returned from `RunPreFilterPlugins`.

Please reorganize stuff based on it.

Reply:
> When all Nodes are filtered out with PreFilterResult, the returning status would be non-success and `diagnosis.AddPluginStatus(s)` at L469 overwrites `UnschedulablePlugins` we set here.

`diagnosis.AddPluginStatus(s)` simply inserts  unschedulable plugin, not overwrites `UnschedulablePlugins` of diagnosis.  the whole logic is as you said above.

Reply:
Oh yeah, right. Never mind.